roles/edpm_nova/defaults/main.yml

```diff
@@ -31,6 +31,11 @@ edpm_nova_compute_image: "quay.io/podified-antelope-centos9/openstack-nova-compu
 
 # certs
 edpm_nova_tls_certs_enabled: "{{ edpm_tls_certs_enabled | default(False) }}"
+# NOTE(sean-k-mooney): nova will use unix sockets for libvirt and ovs communication so we
+# will not need the libvirt or ovs client certs or ca. nova will communicate other services
+# via there api endpoints and will connect to rabbitmq. To support this we will need to trust
+# the general ca root cert.
+edpm_nova_tls_ca_src_dir: /var/lib/openstack/cacerts/nova
```

Comment:
on my deployment it's /var/lib/openstack/cacerts/nova-custom (OpenStackDataPlaneService name). @vakwetu FYI

Reply:
hum that was not ment ot change based on the dataplane service name
i don't think we have access to that in ansible

Reply:
the dataplane-operator doesn't know anything about whether the service name is nova or nova-custom or whatever.   It just knows that there is a dataplane service named X and so it puts its ca bundle under /var/lib/openstack/cacerts/X  or certs and keys (and now ca.certs) under /var/lib/openstack/certs/X. 

We can fix this in two ways.  1) We can add something to the service spec -- something like "mountAlias" that tells the dataplane-operator to use a different path instead.  2) we can add an ansible param that specifies the service name

Reply:
i tought about both 2 would only work if differnt named service shared the same tls ca

i.e. if two libivrt serivce had the same ca

i dont think you can do that automatically

so instead of a mount alias i think we need to extend the dataplane service with a type field.
then use that to generate the crts and deifne the mount point.

so type=ovs, type=libvirt or type=nova
that woudl map to 
type=nova with name=nova-custom woudl result in the tls certs beign placed in .../nova ignoring the service name "nova-custom"

/var/lib/openstack/cacerts

do we think that is feisable?

Reply:
So, edpm-nova handles this discrepancy between nova and nova-custom by discovering the config files https://github.com/openstack-k8s-operators/edpm-ansible/blob/f98ed5f09467620a00e2936c27a80120b634a56c/roles/edpm_nova/tasks/configure.yml#L70  and copying them to the right place on the compute node.

install-certs has no such logic.  We just copy the right certs, keys, cacerts for the node from /var/lib/openstack/cacerts in the ansibleEE pod to /var/lib/openstack/cacerts on the compute node.  Same for /var/lib/openstack/certs.

So, we can solve this either by mounting the certs in the ansibleEE container to the right place (/var/lib/openstack/cacerts/nova) to begin with -- by using the mountAlias thing I just mentioned.

Or we can add an ansible param that provides the service-name and then have the edpm-nova role pick things up from the right path.

Which do you guys prefer?

Reply:
gah - didn't refresh to see your response .. reading ..

Reply:
@SeanMooney OK - so I think we're considering something like option 1 - where we have a field that redefines where the certs and cacerts are mounted in the ansibleEE pod to begin with.

The difference is that you call the attribute "type" - which might be useful later in other contexts too - maybe to later simplify the configMap mounts for instance.

The implicit assumption of course is that there is only one dataplane service defined with each "type".

Yeah, this is feasible.  I'll start work on it and get something up for review by early next week.

Reply:
yes so i briefly ran the type idea past gibi and his reaction is that would allow use to add validating/default webhooks for the standard datapalne services to be able to check for example that if you define a dataplane service of type=nova tha the secrete always have a migration ssh key defined.

it would also allow use to enforce that for any node set we only have one datapalneservice for any given type.

the value of type for the standard service would basically map to the last component of the playbook

https://github.com/openstack-k8s-operators/nova-operator/blob/main/api/v1beta1/nova_webhook.go#L74-L139

so type=nova is related to   playbook: osp.edpm.nova

we don't have to enforce that but if we are wondering what the type would be for the exiting service that's what i would use.

Reply:
I don't think option 1 is correct. If we have 2 libvirt services (not sure what the use-case would be but hypothetically....) then they have independent config. If they happen to be using the same CA that's fine, but they should get individual keys/certs/secrets

Reply:
seems to work https://github.com/openstack-k8s-operators/dataplane-operator/commit/dda1a9d0e8d99d616db05a5ff9d9d242ef33f6bf with https://github.com/olliewalsh/edpm-ansible/commit/13b1158ae14ca5cbcd5e327c7d6833906820dc53

Reply:
the service name can solve this problem for this specific usecase so i updated the patch to support that.

i think have a type on the dataplane service so we can use that for validation or defaulting is still worth considering but that can be done separately.

@olliewalsh are you going to submit a pr to the dataplane operator for that.

Reply:
Included in the OVN dataplane PR https://github.com/openstack-k8s-operators/dataplane-operator/pull/745